packages/adblocker/test/engine/engine.test.ts

```diff
@@ -390,13 +401,29 @@ $csp=baz,domain=bar.com
       url: 'https://foo.com',
     });
 
-    const createEngineWithResource = (filters: string[], resource: string) => {
+    const createEngineWithResource = (filters: string[], content: string) => {
       const engine = createEngine(filters.join('\n'));
-      engine.resources.js.set(resource, resource);
-      engine.resources.resources.set(resource, {
-        body: resource,
-        contentType: 'application/javascript',
-      });
+      engine.resources = Resources.parse(
+        JSON.stringify({
+          redirects: [
+            {
+              names: [content],
+              content: content,
+              contentType: 'application/javascript',
+            },
+          ],
+          scriptlets: [
+            {
+              names: [content],
+              content,
+              dependencies: [],
+            },
+          ],
+        } as ResourcesDistribution),
```

Comment:
maybe it's better to put this object on a property typed variable as we know `as` syntax does not guarantee type completeness